services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/Wrappers.groovy

```diff
@@ -99,5 +95,12 @@ class Wrappers {
     }
 
     @InheritConstructors
-    static class WaitTimeoutException extends RuntimeException {}
+    static class WaitTimeoutException extends RuntimeException {
+        Throwable originalError
+        
+        WaitTimeoutException(String message, Throwable error) {
+            super(error ? message + "\nFailed with exception:\n\n$error" : message)
```

Comment:
Your previous variant is more cute I think. Let's leave it. 

super(error ? message + ". Failed with exception:\n\n$error" : message)

Reply:
I don't think it is. Why would we force a dot if there are no guarantees that the `message` won't have its own dot at the end. Moreover, I don't really understand why it matter so much

Reply:
This is not how designing classes work. From the perspective of this constructor, it takes `message` as an external parameter and it should be considered that it knows nothing about those who call it

Reply:
Maybe it is worth showing the original exception with tabs. It will be looked more pretty. For example, we can do it something like that:
super(error ? message + "\n\nFailed with exception:\n\n\t${error.toString().replaceAll("\n", "\n\t")}" : message)

Reply:
In principle, my suggestion is up to you.